src/index.js

```diff
@@ -119,7 +119,7 @@ export default function({ types: t, template }) {
     CallExpression(path) {
       if (
         path.node[VISITED_KEY] ||
-        !matchesPatterns(path.get('callee'), this.factoryMethods) &&
+        !matchesPatterns(path.get('callee'), this.factoryMethods) ||
```

Comment:
if both `matchesPatterns` and `isReactLikeComponentObject` are independent checks for "is this node a react component", then we should only bail and return early if *both* the checks fail, right? if a component would pass the `isReactLikeComponentObject` check but fails the `factoryMethods` check, it never gets a chance to do so:
`if (false || true || false)` -> bail